addons/stock/models/stock_move.py

```diff
@@ -464,18 +466,8 @@ def write(self, vals):
         elif vals.get('state', '') == 'done' and vals.get('date'):
             propagated_date_field = 'date'
         if propagated_date_field:
-            new_date = vals.get(propagated_date_field)
             for move in self:
-                move_dest_ids = move.move_dest_ids.filtered(lambda m: m.state not in ('done', 'cancel'))
-                if not move_dest_ids:
-                    continue
-                delta_days = (new_date - move.date_expected).total_seconds() / 86400
-                if not move.propagate_date or abs(delta_days) < move.propagate_date_minimum_delta:
-                    move_dest_ids._delay_alert_log_activity('manual', move)
-                    continue
-                for move_dest in move_dest_ids:
-                    move_dest.date_expected += relativedelta.relativedelta(days=delta_days)
-                move_dest_ids._delay_alert_log_activity('auto', move)
+                move._delay_alert_log_activity(vals.get(propagated_date_field))
```

Comment:
before the method was in two part
1) reschedule
2) delay alert

now you do the rescheduling in the delay alert method?
also i don't think this should be common: the rescheduling work on the next move, the delay alert has a new special smart logic which is more complex